--- templates/crds/pgreplicas-crd.yaml	2022-01-07 09:09:23.832920142 -0600
+++ templates/crds/pgreplicas-crd.yaml.tmp	2022-01-07 09:09:59.140929232 -0600
@@ -1,8 +1,11 @@
----
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "pgreplicas.pg.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: pgreplicas.pg.percona.com
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
 spec:
   group: pg.percona.com
   names:
@@ -27,3 +30,4 @@
             properties:
               state: { type: string }
               message: { type: string }
+{{- end }}
