--- templates/operator.yaml	2022-08-12 03:51:59.904019670 -0500
+++ templates/operator.yaml.tmp	2022-08-12 03:51:05.976145546 -0500
@@ -3,10 +3,14 @@
 metadata:
   name: pgo-deployer-sa
 ---
+{{- if (not (lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "pgo-deployer-cr")) }}
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: pgo-deployer-cr
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '5'
 rules:
   - apiGroups:
       - ''
@@ -116,6 +120,7 @@
     verbs:
       - delete
       - list
+{{- end }}
 ---
 apiVersion: v1
 kind: ConfigMap
@@ -147,7 +152,7 @@
     ccp_image_pull_secret_manifest: ""
     ccp_image_tag: "1.3.0-ppg14-postgres-ha"
     create_rbac: "true"
-    crunchy_debug: "false"
+    crunchy_debug: {{ .Values.config.debug | quote }}
     db_name: ""
     db_password_age_days: "0"
     db_password_length: "24"
@@ -165,8 +170,8 @@
     reconcile_rbac: "true"
     exporterport: "9187"
     metrics: "false"
-    namespace: "pgo"
-    namespace_mode: "disabled"
+    namespace: {{ .Release.Namespace | quote }}
+    namespace_mode: {{ .Values.config.namespaceMode | quote }}
     pgbadgerport: "10000"
     pgo_add_os_ca_store: "false"
     pgo_admin_password: "examplepassword"
@@ -182,14 +187,14 @@
     pgo_cluster_admin: "false"
     pgo_disable_eventing: "false"
     pgo_disable_tls: "false"
-    pgo_image_prefix: "percona/percona-postgresql-operator"
+    pgo_image_prefix: {{ .Values.images.pgo.repository | quote }}
     pgo_image_pull_policy: "Always"
     pgo_image_pull_secret: ""
     pgo_image_pull_secret_manifest: ""
-    pgo_image_tag: "1.3.0"
+    pgo_image_tag: {{ .Values.images.pgo.tag | quote }}
     pgo_installation_name: "devtest"
     pgo_noauth_routes: ""
-    pgo_operator_namespace: "pgo"
+    pgo_operator_namespace: {{ .Release.Namespace | quote }}
     pgo_tls_ca_store: ""
     pgo_tls_no_verify: "false"
     pod_anti_affinity: "preferred"
@@ -252,7 +257,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: pgo-deployer-crb
+  name: pgo-deployer-crb-{{ .Release.Namespace }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -260,7 +265,7 @@
 subjects:
   - kind: ServiceAccount
     name: pgo-deployer-sa
-    namespace: pgo
+    namespace: {{ .Release.Namespace | quote }}
 ---
 apiVersion: batch/v1
 kind: Job
@@ -276,7 +281,7 @@
       restartPolicy: Never
       containers:
         - name: pgo-deploy
-          image: percona/percona-postgresql-operator:1.3.0-pgo-deployer
+          image: {{ .Values.images.deployer.repository }}:{{ .Values.images.deployer.tag }}
           imagePullPolicy: Always
           env:
             - name: DEPLOY_ACTION
