--- templates/operator.yaml	2022-01-07 11:17:21.954531131 -0600
+++ templates/operator.yaml.tmp	2022-01-07 11:17:41.074714144 -0600
@@ -2,12 +2,15 @@
 kind: ServiceAccount
 metadata:
   name: pgo-deployer-sa
-  namespace: pgo
 ---
+{{- if (not (lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "pgo-deployer-cr")) }}
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: pgo-deployer-cr
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '5'
 rules:
   - apiGroups:
       - ''
@@ -115,12 +118,12 @@
     verbs:
       - delete
       - list
+{{- end }}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: pgo-deployer-cm
-  namespace: pgo
 data:
   values.yaml: |-
     # =====================
@@ -147,7 +150,7 @@
     ccp_image_pull_secret_manifest: ""
     ccp_image_tag: "1.1.0-postgres-ha"
     create_rbac: "true"
-    crunchy_debug: "false"
+    crunchy_debug: {{ .Values.config.debug | quote }}
     db_name: ""
     db_password_age_days: "0"
     db_password_length: "24"
@@ -165,8 +168,8 @@
     reconcile_rbac: "true"
     exporterport: "9187"
     metrics: "false"
-    namespace: "pgo"
-    namespace_mode: "dynamic"
+    namespace: {{ .Release.Namespace | quote }}
+    namespace_mode: "disabled"
     pgbadgerport: "10000"
     pgo_add_os_ca_store: "false"
     pgo_admin_password: "examplepassword"
@@ -189,7 +192,7 @@
     pgo_image_tag: "1.1.0"
     pgo_installation_name: "devtest"
     pgo_noauth_routes: ""
-    pgo_operator_namespace: "pgo"
+    pgo_operator_namespace: {{ .Release.Namespace | quote }}
     pgo_tls_ca_store: ""
     pgo_tls_no_verify: "false"
     pod_anti_affinity: "preferred"
@@ -252,7 +255,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: pgo-deployer-crb
+  name: pgo-deployer-crb-{{ .Release.Namespace }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -260,13 +263,12 @@
 subjects:
   - kind: ServiceAccount
     name: pgo-deployer-sa
-    namespace: pgo
+    namespace: {{ .Release.Namespace | quote }}
 ---
 apiVersion: batch/v1
 kind: Job
 metadata:
   name: pgo-deploy
-  namespace: pgo
 spec:
   backoffLimit: 0
   template:
@@ -277,7 +279,7 @@
       restartPolicy: Never
       containers:
         - name: pgo-deploy
-          image: percona/percona-postgresql-operator:1.1.0-pgo-deployer
+          image: {{ .Values.images.postgresOperator.repository }}:{{ .Values.images.postgresOperator.tag }}
           imagePullPolicy: Always
           env:
             - name: DEPLOY_ACTION
