--- templates/operator.yaml	2022-01-07 09:09:23.832920142 -0600
+++ templates/operator.yaml.tmp	2022-01-07 09:04:37.764844711 -0600
@@ -1,13 +1,22 @@
+{{- if (not (lookup "v1" "ServiceAccount" .Values.namespace "pgo-deployer-sa")) }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: pgo-deployer-sa
-  namespace: pgo
+  namespace: {{ .Values.namespace }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
+{{- end }}
 ---
+{{- if (not (lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "pgo-deployer-cr")) }}
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: pgo-deployer-cr
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
 rules:
   - apiGroups:
       - ''
@@ -115,12 +124,17 @@
     verbs:
       - delete
       - list
+{{- end }}
 ---
+{{- if (not (lookup "v1" "ConfigMap" .Values.namespace "pgo-deployer-cm")) }}
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: pgo-deployer-cm
-  namespace: pgo
+  namespace: {{ .Values.namespace }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
 data:
   values.yaml: |-
     # =====================
@@ -165,7 +179,7 @@
     reconcile_rbac: "true"
     exporterport: "9187"
     metrics: "false"
-    namespace: "pgo"
+    namespace: ""
     namespace_mode: "dynamic"
     pgbadgerport: "10000"
     pgo_add_os_ca_store: "false"
@@ -189,7 +203,7 @@
     pgo_image_tag: "1.1.0"
     pgo_installation_name: "devtest"
     pgo_noauth_routes: ""
-    pgo_operator_namespace: "pgo"
+    pgo_operator_namespace: "{{ .Values.namespace }}"
     pgo_tls_ca_store: ""
     pgo_tls_no_verify: "false"
     pod_anti_affinity: "preferred"
@@ -248,11 +262,16 @@
     storage9_size: "1Gi"
     storage9_type: "dynamic"
     storage9_class: "rook-ceph-block"
+{{- end }}
 ---
+{{- if (not (lookup "rbac.authorization.k8s.io/v1" "ClusterRoleBinding" "" (printf "pgo-deployer-crb-%s" .Values.namespace))) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: pgo-deployer-crb
+  name: pgo-deployer-crb-{{ .Values.namespace }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -260,13 +279,18 @@
 subjects:
   - kind: ServiceAccount
     name: pgo-deployer-sa
-    namespace: pgo
+    namespace: {{ .Values.namespace }}
+{{- end }}
 ---
+{{- if (not (lookup "batch/v1" "Job" .Values.namespace "pgo-deploy")) }}
 apiVersion: batch/v1
 kind: Job
 metadata:
   name: pgo-deploy
-  namespace: pgo
+  namespace: {{ .Values.namespace }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '0'
 spec:
   backoffLimit: 0
   template:
@@ -277,7 +301,7 @@
       restartPolicy: Never
       containers:
         - name: pgo-deploy
-          image: percona/percona-postgresql-operator:1.1.0-pgo-deployer
+          image: {{ .Values.images.postgresOperator.repository }}:{{ .Values.images.postgresOperator.tag }}
           imagePullPolicy: Always
           env:
             - name: DEPLOY_ACTION
@@ -288,4 +312,5 @@
       volumes:
         - name: deployer-conf
           configMap:
-            name: pgo-deployer-cm
\ No newline at end of file
+            name: pgo-deployer-cm
+{{- end }}
