--- templates/crds.yaml	2021-12-26 09:24:26.850505976 -0600
+++ templates/crds.yaml.tmp	2021-12-26 09:25:20.388446205 -0600
@@ -1,3 +1,4 @@
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "perconaxtradbclusters.pxc.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -450,7 +451,9 @@
           specReplicasPath: .spec.pxc.size
           statusReplicasPath: .status.pxc.ready
           labelSelectorPath: .status.pxc.labelSelectorPath
+{{- end }}
 ---
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "perconaxtradbclusterbackups.pxc.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -506,7 +509,9 @@
         jsonPath: .metadata.creationTimestamp
       subresources:
         status: {}
+{{- end }}
 ---
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "perconaxtradbclusterrestores.pxc.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -554,7 +559,9 @@
         jsonPath: .metadata.creationTimestamp
       subresources:
         status: {}
+{{- end }}
 ---
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "perconaxtradbbackups.pxc.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -606,178 +613,4 @@
       - name: Age
         type: date
         jsonPath: .metadata.creationTimestamp
----
-kind: Role
-apiVersion: rbac.authorization.k8s.io/v1
-metadata:
-  name: percona-xtradb-cluster-operator
-rules:
-- apiGroups:
-  - pxc.percona.com
-  resources:
-  - perconaxtradbclusters
-  - perconaxtradbclusters/status
-  - perconaxtradbclusterbackups
-  - perconaxtradbclusterbackups/status
-  - perconaxtradbclusterrestores
-  - perconaxtradbclusterrestores/status
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-- apiGroups:
-  - ""
-  resources:
-  - pods
-  - pods/exec
-  - pods/log
-  - configmaps
-  - services
-  - persistentvolumeclaims
-  - secrets
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-- apiGroups:
-  - apps
-  resources:
-  - deployments
-  - replicasets
-  - statefulsets
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-- apiGroups:
-  - batch
-  resources:
-  - jobs
-  - cronjobs
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-- apiGroups:
-  - policy
-  resources:
-  - poddisruptionbudgets
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-- apiGroups:
-  - certmanager.k8s.io
-  - cert-manager.io
-  resources:
-  - issuers
-  - certificates
-  verbs:
-  - get
-  - list
-  - watch
-  - create
-  - update
-  - patch
-  - delete
-  - deletecollection
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: percona-xtradb-cluster-operator
----
-kind: RoleBinding
-apiVersion: rbac.authorization.k8s.io/v1
-metadata:
-  name: service-account-percona-xtradb-cluster-operator
-subjects:
-- kind: ServiceAccount
-  name: percona-xtradb-cluster-operator
-roleRef:
-  kind: Role
-  name: percona-xtradb-cluster-operator
-  apiGroup: rbac.authorization.k8s.io
----
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: percona-xtradb-cluster-operator
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app.kubernetes.io/component: operator
-      app.kubernetes.io/instance: percona-xtradb-cluster-operator
-      app.kubernetes.io/name: percona-xtradb-cluster-operator
-      app.kubernetes.io/part-of: percona-xtradb-cluster-operator
-  strategy:
-    rollingUpdate:
-      maxUnavailable: 1
-    type: RollingUpdate
-  template:
-    metadata:
-      labels:
-        app.kubernetes.io/component: operator
-        app.kubernetes.io/instance: percona-xtradb-cluster-operator
-        app.kubernetes.io/name: percona-xtradb-cluster-operator
-        app.kubernetes.io/part-of: percona-xtradb-cluster-operator
-    spec:
-      terminationGracePeriodSeconds: 600
-      containers:
-      - command:
-        - percona-xtradb-cluster-operator
-        env:
-        - name: WATCH_NAMESPACE
-          valueFrom:
-            fieldRef:
-              apiVersion: v1
-              fieldPath: metadata.namespace
-        - name: POD_NAME
-          valueFrom:
-            fieldRef:
-              apiVersion: v1
-              fieldPath: metadata.name
-        - name: OPERATOR_NAME
-          value: percona-xtradb-cluster-operator
-        image: percona/percona-xtradb-cluster-operator:1.10.0
-        imagePullPolicy: Always
-        livenessProbe:
-          failureThreshold: 3
-          httpGet:
-            path: /metrics
-            port: metrics
-            scheme: HTTP
-        resources:
-          limits:
-            cpu: 200m
-            memory: 500Mi
-          requests:
-            cpu: 100m
-            memory: 20Mi
-        name: percona-xtradb-cluster-operator
-        ports:
-        - containerPort: 8080
-          name: metrics
-          protocol: TCP
-      serviceAccountName: percona-xtradb-cluster-operator
+{{- end }}
