--- templates/crds/pgclusters-crd.yaml	2022-01-07 09:19:11.245067732 -0600
+++ templates/crds/pgclusters-crd.yaml.tmp	2022-01-07 09:18:49.705062412 -0600
@@ -1,8 +1,11 @@
----
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "pgclusters.pg.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: pgclusters.pg.percona.com
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '1'
 spec:
   group: pg.percona.com
   names:
@@ -37,3 +40,4 @@
             properties:
               state: { type: string }
               message: { type: string }
+{{- end }}
