--- templates/olm.yaml	2023-06-17 09:25:18.670172018 -0400
+++ templates/olm.yaml.tmp	2023-06-17 09:24:29.663069017 -0400
@@ -2,11 +2,6 @@
 apiVersion: v1
 kind: Namespace
 metadata:
-  name: olm
----
-apiVersion: v1
-kind: Namespace
-metadata:
   name: operators
 ---
 kind: ServiceAccount
@@ -66,9 +61,11 @@
           args:
           - --namespace
           - $(OPERATOR_NAMESPACE)
+          - --opmImage
+          - "{{ .Values.images.olm.repository }}:{{ .Values.images.olm.tag }}"
           - --writeStatusName
           - ""
-          image: quay.io/operator-framework/olm@sha256:e74b2ac57963c7f3ba19122a8c31c9f2a0deb3c0c5cac9e5323ccffd0ca198ed
+          image: "{{ .Values.images.olm.repository }}:{{ .Values.images.olm.tag }}"
           imagePullPolicy: IfNotPresent
           ports:
             - containerPort: 8080
