--- templates/olm.yaml	2023-06-17 10:53:34.951035808 -0400
+++ templates/olm.yaml.tmp	2023-06-17 10:55:32.535713372 -0400
@@ -2,18 +2,6 @@
 apiVersion: v1
 kind: Namespace
 metadata:
-  name: olm
-  labels:
-    pod-security.kubernetes.io/enforce: restricted
-    pod-security.kubernetes.io/enforce-version: latest
-    pod-security.kubernetes.io/audit: restricted
-    pod-security.kubernetes.io/audit-version: latest
-    pod-security.kubernetes.io/warn: restricted
-    pod-security.kubernetes.io/warn-version: latest
----
-apiVersion: v1
-kind: Namespace
-metadata:
   name: operators
   labels:
     pod-security.kubernetes.io/enforce: baseline
