--- templates/crds/keycloakbackups.keycloak.org.crd.yaml	2022-10-07 06:43:07.797085294 -0500
+++ templates/crds/keycloakbackups.keycloak.org.crd.yaml.tmp	2022-10-07 06:46:02.702945189 -0500
@@ -1,7 +1,11 @@
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "keycloakbackups.keycloak.org")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: keycloakbackups.keycloak.org
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: '5'
 spec:
   group: keycloak.org
   names:
@@ -151,3 +155,4 @@
     storage: true
     subresources:
       status: {}
+{{- end }}
