--- templates/operator.yaml	2022-03-10 23:42:10.063315505 +0530
+++ templates/operator.yaml.tmp	2022-03-10 23:53:37.055556734 +0530
@@ -1,3 +1,5 @@
+{{- $existingCrd := lookup "apiextensions.k8s.io/v1beta1" "CustomResourceDefinition" "" "gitlabs.apps.gitlab.com" }}
+{{- if (not $existingCrd) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -116,6 +118,7 @@
     plural: ""
   conditions: []
   storedVersions: []
+{{- end }}
 ---
 apiVersion: v1
 kind: ServiceAccount
@@ -239,6 +242,8 @@
   - create
   - patch
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "gitlab-app-role" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
@@ -253,7 +258,10 @@
   - securitycontextconstraints
   verbs:
   - use
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "gitlab-manager-role" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
@@ -526,7 +534,11 @@
   - patch
   - update
   - watch
+
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "gitlab-metrics-reader" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
@@ -536,7 +548,10 @@
   - /metrics
   verbs:
   - get
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "gitlab-nginx-ingress" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
@@ -596,7 +611,10 @@
   - get
   - list
   - watch
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRole" "" "gitlab-proxy-role" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
@@ -614,6 +632,7 @@
   - subjectaccessreviews
   verbs:
   - create
+{{- end }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
@@ -643,6 +662,8 @@
   name: gitlab-nginx-ingress
   namespace: gitlab-system
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRoleBinding" "" "gitlab-gitlab-app-rolebinding" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
@@ -655,7 +676,10 @@
 - kind: ServiceAccount
   name: gitlab-app
   namespace: gitlab-system
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRoleBinding" "" "gitlab-gitlab-manager-rolebinding" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
@@ -668,7 +692,10 @@
 - kind: ServiceAccount
   name: gitlab-manager
   namespace: gitlab-system
+{{- end}}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRoleBinding" "" "gitlab-gitlab-nginx-ingress" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
@@ -681,7 +708,10 @@
 - kind: ServiceAccount
   name: gitlab-nginx-ingress
   namespace: gitlab-system
+{{- end }}
 ---
+{{- $existing := lookup "rbac.authorization.k8s.io/v1" "ClusterRoleBinding" "" "gitlab-gitlab-proxy-rolebinding" }}
+{{- if (not $existing) }}
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
@@ -694,6 +724,7 @@
 - kind: ServiceAccount
   name: gitlab-manager
   namespace: gitlab-system
+{{- end }}
 ---
 apiVersion: v1
 kind: Service
@@ -832,6 +863,8 @@
 spec:
   controller: k8s.io/ingress-nginx
 ---
+{{- $existing := lookup "admissionregistration.k8s.io/v1" "ValidatingWebhookConfiguration" "" "gitlab-gitlab-validating-webhook-configuration" }}
+{{- if (not $existing) }}
 apiVersion: admissionregistration.k8s.io/v1
 kind: ValidatingWebhookConfiguration
 metadata:
@@ -859,3 +892,4 @@
     resources:
     - gitlabs
   sideEffects: None
+{{- end }}
\ No newline at end of file
