--- templates/operator.yaml	2021-12-24 08:40:04.260983308 -0600
+++ templates/operator.yaml.tmp	2021-12-24 08:42:47.915464310 -0600
@@ -1,9 +1,14 @@
+{{- $existingCrd := lookup "apiextensions.k8s.io/v1beta1" "CustomResourceDefinition" "" "gitlabs.apps.gitlab.com" }}
+{{- if (not $existingCrd) }}
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    cert-manager.io/inject-ca-from: gitlab-system/gitlab-serving-cert
+    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/gitlab-serving-cert
     controller-gen.kubebuilder.io/version: v0.3.0
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
   name: gitlabs.apps.gitlab.com
 spec:
   additionalPrinterColumns:
@@ -118,32 +123,33 @@
     plural: ""
   conditions: []
   storedVersions: []
+{{- end }}
 ---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   creationTimestamp: null
   name: gitlab-app
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   creationTimestamp: null
   name: gitlab-manager
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: gitlab-nginx-ingress
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   name: gitlab-leader-election-role
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 rules:
 - apiGroups:
   - ""
@@ -171,7 +177,7 @@
 kind: Role
 metadata:
   name: gitlab-nginx-ingress
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 rules:
 - apiGroups:
   - ""
@@ -257,6 +263,10 @@
 metadata:
   creationTimestamp: null
   name: gitlab-app-role
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 rules:
 - apiGroups:
   - security.openshift.io
@@ -272,6 +282,10 @@
 metadata:
   creationTimestamp: null
   name: gitlab-manager-role
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 rules:
 - apiGroups:
   - apps
@@ -544,6 +558,10 @@
 kind: ClusterRole
 metadata:
   name: gitlab-proxy-role
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 rules:
 - apiGroups:
   - authentication.k8s.io
@@ -562,6 +580,10 @@
 kind: ClusterRole
 metadata:
   name: gitlab-metrics-reader
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 rules:
 - nonResourceURLs:
   - /metrics
@@ -572,7 +594,7 @@
 kind: RoleBinding
 metadata:
   name: gitlab-leader-election-rolebinding
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
@@ -580,13 +602,13 @@
 subjects:
 - kind: ServiceAccount
   name: gitlab-manager
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
   name: gitlab-nginx-ingress
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
@@ -594,12 +616,16 @@
 subjects:
 - kind: ServiceAccount
   name: gitlab-nginx-ingress
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: gitlab-gitlab-app-rolebinding
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -607,12 +633,16 @@
 subjects:
 - kind: ServiceAccount
   name: gitlab-app
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: gitlab-gitlab-manager-rolebinding
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -620,12 +650,16 @@
 subjects:
 - kind: ServiceAccount
   name: gitlab-manager
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: gitlab-gitlab-proxy-rolebinding
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -633,7 +667,7 @@
 subjects:
 - kind: ServiceAccount
   name: gitlab-manager
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 ---
 apiVersion: v1
 kind: Service
@@ -641,7 +675,7 @@
   labels:
     control-plane: controller-manager
   name: gitlab-controller-manager-metrics-service
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 spec:
   ports:
   - name: https
@@ -654,7 +688,7 @@
 kind: Service
 metadata:
   name: gitlab-webhook-service
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 spec:
   ports:
   - port: 443
@@ -669,7 +703,7 @@
   labels:
     control-plane: controller-manager
   name: gitlab-controller-manager
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 spec:
   replicas: 1
   selector:
@@ -687,11 +721,13 @@
         - --zap-devel=true
         command:
         - /manager
+{{- if .Values.config.watchNamespace }}
         env:
         - name: WATCH_NAMESPACE
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+{{- end }}
         image: registry.gitlab.com/gitlab-org/cloud-native/gitlab-operator:0.2.0
         imagePullPolicy: Always
         livenessProbe:
@@ -743,25 +779,25 @@
           defaultMode: 420
           secretName: webhook-server-cert
 ---
-apiVersion: cert-manager.io/v1alpha2
+apiVersion: cert-manager.io/v1
 kind: Certificate
 metadata:
   name: gitlab-serving-cert
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 spec:
   dnsNames:
-  - gitlab-webhook-service.gitlab-system.svc
-  - gitlab-webhook-service.gitlab-system.svc.cluster.local
+  - gitlab-webhook-service.{{ .Release.Namespace }}.svc
+  - gitlab-webhook-service.{{ .Release.Namespace }}.svc.cluster.local
   issuerRef:
     kind: Issuer
     name: gitlab-selfsigned-issuer
   secretName: webhook-server-cert
 ---
-apiVersion: cert-manager.io/v1alpha2
+apiVersion: cert-manager.io/v1
 kind: Issuer
 metadata:
   name: gitlab-selfsigned-issuer
-  namespace: gitlab-system
+  namespace: {{ .Release.Namespace }}
 spec:
   selfSigned: {}
 ---
@@ -769,14 +805,17 @@
 kind: ValidatingWebhookConfiguration
 metadata:
   annotations:
-    cert-manager.io/inject-ca-from: gitlab-system/gitlab-serving-cert
+    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/gitlab-serving-cert
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: '0'
+    helm.sh/hook-delete-policy: before-hook-creation
   name: gitlab-gitlab-validating-webhook-configuration
 webhooks:
 - clientConfig:
     caBundle: Cg==
     service:
       name: gitlab-webhook-service
-      namespace: gitlab-system
+      namespace: {{ .Release.Namespace }}
       path: /validate-apps-gitlab-com-v1beta1-gitlab
   failurePolicy: Fail
   name: vgitlab.kb.io
