--- templates/crds/pgpolicies-crd.yaml	2022-06-30 01:51:08.543792939 +0530
+++ templates/crds/pgpolicies-crd.yaml.tmp	2022-06-30 01:59:24.702360381 +0530
@@ -1,4 +1,5 @@
 ---
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "pgpolicies.pg.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -28,3 +29,4 @@
             properties:
               state: { type: string }
               message: { type: string }
+{{- end }}
\ No newline at end of file
