--- templates/crds/pgtasks-crd.yaml	2022-06-30 01:51:08.671792004 +0530
+++ templates/crds/pgtasks-crd.yaml.tmp	2022-06-30 02:00:17.350339652 +0530
@@ -1,4 +1,5 @@
 ---
+{{- if (not (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "pgtasks.pg.percona.com")) }}
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -29,3 +30,4 @@
             properties:
               state: { type: string }
               message: { type: string }
+{{- end }}
\ No newline at end of file
